lib/ReactViews/Tools/DeltaTool/DeltaTool.jsx

```diff
@@ -0,0 +1,221 @@
+    if (primaryDate) {
+      setPrimaryDate(primaryDate);
+      setSecondaryDate(secondaryDate || primaryDate);
+    }
+  }
+
+  return (
+    <div className={Styles.deltaTool}>
+      <h1 className={Styles.title}>Change Detection: {catalogItem.name}</h1>
+      <div className={Styles.body}>
+        <div>
+          <span>
+            This tool visualizes the difference between imagery captured at two
+            discrete points in time.
+          </span>
+          {location && <PrettyLocation location={location} />}
+          {location === undefined ? (
+            <h3>
+              To view available imagery, please select your location of interest
+              on the map opposite.
+            </h3>
+          ) : (
+            <DatePickers
+              item={item}
+              primaryDate={primaryDate}
+              setPrimaryDate={setPrimaryDate}
+              secondaryDate={secondaryDate}
+              setSecondaryDate={setSecondaryDate}
+            />
+          )}
+        </div>
+        <div className={Styles.buttons}>
+          <button className={Styles.cancelBtn} onClick={cancelDeltaTool}>
+            Cancel
+          </button>
+          <button
+            className={Styles.generateDeltaBtn}
+            onClick={generateDelta}
+            disabled={location === undefined}
+          >
+            Generate Difference Map
+          </button>
+        </div>
+      </div>
+      <LocationPicker
+        terria={terria}
+        location={location}
+        onPick={onUserPickLocation}
+        message={pickerMessage}
+      />
+    </div>
+  );
+}
+
+DeltaTool.propTypes = {
+  terria: PropTypes.object.isRequired,
+  tool: PropTypes.object.isRequired,
+  onCloseTool: PropTypes.func.isRequired
+};
+
+DeltaTool.displayName = "DeltaTool";
+
+function PrettyLocation({ location }) {
+  const prettyLocation = prettifyCoordinates(
+    location.longitude,
+    location.latitude
+  );
+  return (
+    <section>
+      <h4>Selected Location</h4>
+      <div className={Styles.location}>
+        {prettyLocation.latitude}, {prettyLocation.longitude}
+      </div>
+    </section>
+  );
+}
+
+PrettyLocation.propTypes = {
+  location: PropTypes.object.isRequired
+};
+
+PrettyLocation.displayName = "PrettyLocation";
+
+function duplicateItem(item) {
+  const serializedItem = item.serializeToJson();
+  serializedItem.name = serializedItem.name + " (copy)";
+  serializedItem.parameters = Object.assign({}, serializedItem.parameters);
+  delete serializedItem.id;
+  const newItem = createCatalogMemberFromType(item.type, item.terria);
+  newItem.updateFromJson(serializedItem);
+  return newItem;
+}
+
+function trimLines(text) {
+  return text
+    .split("\n")
+    .map(ln => ln.trim())
+    .join("\n");
+}
+
+function dateParamFormat(date) {
+  return dateFormat(date, "yyyy-mm-dd'T'HH:MM:ss.l'Z'", true);
+}
+
+function dateDisplayFormat(date) {
+  return dateFormat(date, "dd-mm-yyyy", true);
+}
+
+module.exports = DeltaTool;
```

Comment:
(PS all our JSX files can safely use import/export as it's not used by the doc generation)